src/Fixer/Whitespace/SpacesInsideParenthesisFixer.php

```diff
@@ -0,0 +1,207 @@
+                if (null !== $prevIndex && $tokens[$prevIndex]->isGivenKind(T_ARRAY)) {
+                    continue;
+                }
+
+                $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+
+                // remove space after opening `(`
+                if (!$tokens[$tokens->getNextNonWhitespace($index)]->isComment()) {
+                    $this->removeSpaceAroundToken($tokens, $index + 1);
+                }
+
+                // remove space before closing `)` if it is not `list($a, $b, )` case
+                if (!$tokens[$tokens->getPrevMeaningfulToken($endIndex)]->equals(',')) {
+                    $this->removeSpaceAroundToken($tokens, $endIndex - 1);
+                }
+            }
+        }
+
+        if ('spaces' === $this->configuration['space']) {
+            foreach ($tokens as $index => $token) {
+                if (!$token->equals('(')) {
+                    continue;
+                }
+
+                $endParenthesisIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+
+                // if not other content than spaces in block remove spaces
+                $blockContent = $this->getBlockContent($index, $endParenthesisIndex, $tokens);
+                if (1 === \count($blockContent) && \in_array(' ', $blockContent, true)) {
+                    $this->removeSpaceAroundToken($tokens, $index + 1);
+
+                    continue;
+                }
+
+                // don't process if the next token is `)`
+                $nextMeaningfulTokenIndex = $tokens->getNextMeaningfulToken($index);
+                if (')' === $tokens[$nextMeaningfulTokenIndex]->getContent()) {
+                    continue;
+                }
+
+                $afterParenthesisIndex = $tokens->getNextNonWhitespace($endParenthesisIndex);
+                $afterParenthesisToken = $tokens[$afterParenthesisIndex];
+
+                if ($afterParenthesisToken->isGivenKind(CT::T_USE_LAMBDA)) {
+                    $useStartParenthesisIndex = $tokens->getNextTokenOfKind($afterParenthesisIndex, ['(']);
+                    $useEndParenthesisIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $useStartParenthesisIndex);
+
+                    // add single-line edge whitespaces inside use parentheses
+                    $this->fixParenthesisInnerEdge($tokens, $useStartParenthesisIndex, $useEndParenthesisIndex);
+                }
+
+                // add single-line edge whitespaces inside parameters list parentheses
+                $this->fixParenthesisInnerEdge($tokens, $index, $endParenthesisIndex);
+            }
+        }
+    }
+
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('space', 'Whether to have `spaces` or `none` spaces inside parenthesis.'))
+                ->setAllowedValues(['none', 'spaces'])
```

Comment:
```suggestion
            (new FixerOptionBuilder('space', 'Whether to have `single` or `none` spaces inside parenthesis.'))
                ->setAllowedValues(['none', 'single'])
```

- `space: spaces` configuration looks weird
- it's not configurable, so it's actually one possible spacing (single space)

Reply:
I think `single` is clear enough. I used `spaces` more as a "working name".